src/benchmarks/micro/libraries/System.Runtime/Perf.HashCode.cs

```diff
@@ -41,114 +23,128 @@ public int Add()
             { 
-                DontDiscard(HashCode.Combine(
-                    DontFold(i)));
+                result = HashCode.Combine(
+                    i + result);
             }
+            return result;
         }
 
-
         [Benchmark]
-        public void Combine_2()
+        public int Combine_2()
         {
+            int result = 0;
             for (int i = 0; i < 10000; i++)
             {
-                DontDiscard(HashCode.Combine(
-                    DontFold(i),
-                    DontFold(i)));
+                result = HashCode.Combine(
+                    i + result,
+                    i + result);
             }
+            return result;
         }
 
         [Benchmark]
-        public void Combine_3()
+        public int Combine_3()
         {
+            int result = 0;
             for (int i = 0; i < 10000; i++)
             {
-                DontDiscard(HashCode.Combine(
-                    DontFold(i),
-                    DontFold(i),
-                    DontFold(i)));
+                result = HashCode.Combine(
+                    i + result,
+                    i + result,
+                    i + result);
             }
+            return result;
         }
 
         [Benchmark]
-        public void Combine_4()
+        public int Combine_4()
         {
+            int result = 0;
             for (int i = 0; i < 10000; i++)
             {
-                DontDiscard(HashCode.Combine(
-                    DontFold(i),
-                    DontFold(i),
-                    DontFold(i),
-                    DontFold(i)));
+                result = HashCode.Combine(
+                    i + result,
+                    i + result,
+                    i + result,
+                    i + result);
             }
+            return result;
         }
 
         [Benchmark]
-        public void Combine_5()
+        public int Combine_5()
         {
+            int result = 0;
             for (int i = 0; i < 10000; i++)
             {
-                DontDiscard(HashCode.Combine(
-                    DontFold(i),
-                    DontFold(i),
-                    DontFold(i),
-                    DontFold(i),
-                    DontFold(i)));
+                result = HashCode.Combine(
+                    i + result,
+                    i + result,
+                    i + result,
+                    i + result,
+                    i + result);
             }
+            return result;
         }
 
         [Benchmark]
-        public void Combine_6()
+        public int Combine_6()
         {
+            int result = 0;
             for (int i = 0; i < 10000; i++)
             {
-                DontDiscard(HashCode.Combine(
-                    DontFold(i),
-                    DontFold(i),
-                    DontFold(i),
-                    DontFold(i),
-                    DontFold(i),
-                    DontFold(i)));
+                result = HashCode.Combine(
```

Comment:
Just to be sure, did you confirm that this doesn't have any adverse effect on the intention of the benchmark?

Reply:
I have not compared the produced disassembly but verified that the benchmarks are not optimized to a single call to `HashCode.Combine`. I hope that this is enough (we want to benchmark the particular methods using input that changes with every loop iteration)

BEFORE

|    Method |       Mean |     Error |    StdDev |     Median |        Min |        Max | Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------- |-----------:|----------:|----------:|-----------:|-----------:|-----------:|------:|------:|------:|----------:|
|       Add |   3.688 us | 0.0277 us | 0.0231 us |   3.702 us |   3.659 us |   3.721 us |     - |     - |     - |         - |
| Combine_1 |  68.680 us | 0.3437 us | 0.3215 us |  68.876 us |  68.058 us |  68.981 us |     - |     - |     - |         - |
| Combine_2 |  82.845 us | 0.5915 us | 0.5533 us |  83.093 us |  81.930 us |  83.595 us |     - |     - |     - |         - |
| Combine_3 |  95.124 us | 0.6227 us | 0.5520 us |  95.384 us |  94.127 us |  95.819 us |     - |     - |     - |         - |
| Combine_4 |  99.920 us | 0.7247 us | 0.6779 us | 100.305 us |  98.696 us | 100.681 us |     - |     - |     - |         - |
| Combine_5 | 117.198 us | 0.9800 us | 0.9167 us | 116.894 us | 115.617 us | 118.414 us |     - |     - |     - |         - |
| Combine_6 | 128.352 us | 1.0133 us | 0.8983 us | 128.091 us | 127.247 us | 129.667 us |     - |     - |     - |         - |
| Combine_7 | 142.305 us | 0.8045 us | 0.7525 us | 142.210 us | 140.931 us | 143.547 us |     - |     - |     - |         - |
| Combine_8 | 154.183 us | 1.5598 us | 1.4591 us | 154.196 us | 152.158 us | 156.671 us |     - |     - |     - |         - |

AFTER

|    Method |       Mean |     Error |    StdDev |     Median |        Min |        Max | Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------- |-----------:|----------:|----------:|-----------:|-----------:|-----------:|------:|------:|------:|----------:|
|       Add |   3.698 us | 0.0229 us | 0.0214 us |   3.691 us |   3.672 us |   3.748 us |     - |     - |     - |         - |
| Combine_1 |  55.353 us | 0.3684 us | 0.3446 us |  55.593 us |  54.724 us |  55.711 us |     - |     - |     - |         - |
| Combine_2 |  68.739 us | 0.4063 us | 0.3800 us |  68.869 us |  67.948 us |  69.091 us |     - |     - |     - |         - |
| Combine_3 |  82.176 us | 0.2895 us | 0.2708 us |  82.197 us |  81.258 us |  82.403 us |     - |     - |     - |         - |
| Combine_4 |  76.823 us | 0.5927 us | 0.5544 us |  76.916 us |  75.832 us |  77.733 us |     - |     - |     - |         - |
| Combine_5 |  88.316 us | 0.4261 us | 0.3777 us |  88.414 us |  87.312 us |  88.668 us |     - |     - |     - |         - |
| Combine_6 |  99.235 us | 0.7510 us | 0.7025 us |  98.875 us |  98.552 us | 100.343 us |     - |     - |     - |         - |
| Combine_7 | 114.906 us | 0.8534 us | 0.7982 us | 114.485 us | 114.092 us | 115.988 us |     - |     - |     - |         - |
| Combine_8 | 100.023 us | 0.3219 us | 0.2513 us | 100.094 us |  99.413 us | 100.279 us |     - |     - |     - |         - |

Reply:
Excellent. Thanks for confirming.